### PR TITLE
Feacure/g khyzhynska/crud frontend

### PR DESCRIPTION
**Pull request by Gal4enock**
CRUD operations for mobile, types, for TS, screens

### CODE REVIEW

File: mobile/screen/auth/StartPage.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  StyleSheet,
+  Text,
+} from 'react-native';
+
+const StartPage: React.FC<{}> = ({ navigation }) => {
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity
+        style={styles.customBtnBG}
+        onPress={() => navigation.navigate('TodoList') } >
```
Comment (CodeGeneration-2020) [inline mobile/screen/auth/StartPage.tsx:14]: inline function

File: mobile/screen/auth/StartPage.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        <Text style={styles.customBtnText}>Login</Text>
+      </TouchableOpacity>
+      <TouchableOpacity
+        style={styles.customBtnBG}
+        onPress={() => navigation.navigate('TodoList') } >
```
Comment (CodeGeneration-2020) [inline mobile/screen/auth/StartPage.tsx:19]: inline function

File: mobile/services/TodoModel.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const createTodoModel = (todoFromServer: any) => {
+  return new TodoModel(todoFromServer._id,
+      todoFromServer.title,
+      todoFromServer.description,
+      todoFromServer.year,
+      todoFromServer.isPublic,
+      todoFromServer.isCompleted);
+};
+
+export { createTodoModel };
```
Comment (CodeGeneration-2020) [inline mobile/services/TodoModel.ts:26]: please do line-break below

**Review by CodeGeneration-2020 — CHANGES_REQUESTED** [review]
- Please avoid inline-function all over the app. Everything should be named
- please do proper spacings all over the app. I showed you a few examples
- avoid mistakes in english words

File: mobile/screen/auth/LoginScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            />
+            <Text>Password</Text>
+            <TextInput
+              multiline
+              placeholder="************"
+              onChangeText={handleChange('password')}
+              onBlur={handleBlur('password')}
+              value={values.password}
+            />
+            <Button onPress={() => handleSubmit()} title="Submit" />
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/screen/auth/LoginScreen.tsx:40]: this component is developing still.. and i had an error, writing this function like: onPress={handleSubmit}

File: mobile/screen/auth/RegisterScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            />
+            <Text>Password</Text>
+            <TextInput
+              multiline
+              placeholder="************"
+              onChangeText={handleChange('password')}
+              onBlur={handleBlur('password')}
+              value={values.password}
+            />
+            <Button onPress={() => handleSubmit()} title="Submit" />
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/screen/auth/RegisterScreen.tsx:40]: this component is developing still.. and i had an error, writing this function like: onPress={handleSubmit}

File: mobile/App.tsx [hunk 6]
```diff
@@ -1,20 +1,51 @@
...
+import { NavigationContainer } from '@react-navigation/native';
+import { View, StyleSheet } from 'react-native';
+import { CreateNewTodo } from './components/TodoElement';
+import TodoList from './components/TodoList';
+import StartPage from './screen/auth/StartPage';
+import RegisterScreen from './screen/auth/RegisterScreen';
+import LoginScreen from './screen/auth/LoginScreen';
+import { EditTodo } from './components/EditTodo';
+
+
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/App.tsx]: fixed in another pull request

File: mobile/App.tsx [hunk 7]
```diff
@@ -1,20 +1,51 @@
...
 export default function App() {
   return (
     <View style={styles.container}>
-      <Text>We are Live!</Text>
-      <StatusBar style="auto" />
+      <NavigationContainer>
+        <Stack.Navigator>
+          <Stack.Screen options={{
+            headerShown: false,
+          }} name="StartPage" component={ StartPage }/>
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/App.tsx]: fixed in another pull request

File: mobile/App.tsx [hunk 8]
```diff
@@ -1,20 +1,51 @@
...
+          <Stack.Screen options={{
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/App.tsx]: fixed in another pull request

File: mobile/components/EditTodo.tsx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    <View style={styles.container}>
+      <Text style={styles.titleCreate}>Create new Todo</Text>
+      <Formik
+        initialValues={{
+          title: editedTodo.title,
+          description: editedTodo.description,
+          year: editedTodo.year,
+        }}
+        onSubmit={(values) => handleSubmitForm(values)}>
+        {({ handleChange,
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/EditTodo.tsx:37]: fixed in another pull request

File: mobile/screen/auth/RegisterScreen.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/screen/auth/RegisterScreen.tsx:40]: we've spoke about it

File: mobile/App.tsx [hunk 10]
```diff
@@ -1,20 +1,51 @@
...
 const styles = StyleSheet.create({
   container: {
-    flex: 1,
-    backgroundColor: '#fff',
-    alignItems: 'center',
-    justifyContent: 'center',
+    width: '100%',
+    height: '100%',
+    paddingTop: 10,
+    // alignItems: 'center',
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/App.tsx]: fixed in another pull request

File: mobile/components/EditTodo.tsx [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+            <View style={[styles.checkingBox, styles.spaceBox]}>
+              <Text>Completed</Text>
+              <CheckBox
+                containerStyle={styles.check}
+                checkedColor='black'
+                checked={isCompleted}
+                onPress={togleCompleted}
+              />
+            </View>
+            <Button onPress={() => handleSubmit()} title="Edit" />
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/EditTodo.tsx:81]: fixed in another pull request

File: mobile/components/TodoElement.tsx [hunk 12]
```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import { useState } from 'react';
+import {
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoElement.tsx:3]: fixed in another pull request

File: mobile/components/TodoElement.tsx [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+    <View style={styles.container}>
+      <Text style={styles.titleCreate}>Create new Todo</Text>
+      <Formik
+        initialValues={{
+          title: '',
+          description: '',
+          year: 2021,
+        }}
+        onSubmit={(values) => handleSubmitForm(values)}>
+        {({ handleChange,
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoElement.tsx:40]: fixed in another pull request

File: mobile/components/TodoElement.tsx [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+              }}
+            >
+              <Text>Create</Text>
+            </TouchableOpacity>
+          </View>
+        )}
+      </Formik>
+    </View>
+  );
+};
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoElement.tsx:100]: fixed in another pull request

File: mobile/components/TodoList.tsx [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+  _id: string;
+  description: string;
+  year: number;
+  isPublic: boolean;
+  isComplited: boolean;
+  handleDelete: Function;
+  handleEdit: Function;
+};
+
+const Item = ({ title, _id, description, year,
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoList.tsx:22]: fixed in another pull request

File: mobile/components/TodoList.tsx [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+
+const TodoList = ({ navigation }) => {
+  const [todoList, settodoList] = useState([]);
+  useEffect(() => {
+    const gettodoList = async () => {
+      const allTodos = await todoService.getTodo();
+      settodoList(allTodos);
+    };
+    gettodoList();
+    console.log(todoList);
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoList.tsx:80]: fixed in another pull request

File: mobile/components/TodoList.tsx [hunk 17]
```diff
@@ -0,0 +1,186 @@
...
+  }, []);
+
+  const renderItem = ({ item }) => (
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoList.tsx:83]: fixed in another pull request

File: mobile/components/TodoList.tsx [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+      name: 'EditTodo',
+      params: { todo },
+    });
+  };
+
+  return (
+    <View style={styles.container}>
+      <TouchableOpacity
+        style={styles.button}
+        onPress={() => {
```
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoList.tsx:109]: fixed in another pull request

File: mobile/components/TodoList.tsx (hunk 17 above)
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/components/TodoList.tsx:83]: fixed in another pull request

File: mobile/screen/auth/LoginScreen.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Gal4enock) [reply mobile/screen/auth/LoginScreen.tsx:40]: fixed in another pull request

File: mobile/screen/auth/StartPage.tsx (hunk 1 above)
Comment (Gal4enock) [reply mobile/screen/auth/StartPage.tsx]: fixed in another pull request

File: mobile/screen/auth/StartPage.tsx (hunk 2 above)
Comment (Gal4enock) [reply mobile/screen/auth/StartPage.tsx]: fixed in another pull request

File: mobile/services/TodoModel.ts (hunk 3 above)
Comment (Gal4enock) [reply mobile/services/TodoModel.ts:26]: fixed in another pull request

**Comment by Gal4enock** [comment]
please, check pull https://github.com/CodeGeneration-2020/CGS-test-camp-task/pull/109

**Pull request closed without merging**